contracts/contracts/token/OUSD.sol

```diff
@@ -504,6 +504,8 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
     function rebaseOptIn() public nonReentrant {
         require(_isNonRebasingAccount(msg.sender), "Account has not opted out");
 
+        uint256 oldBalance = balanceOf(msg.sender);
```

Comment:
maybe add a comment here as to why it is important that the balance is of a user to be subtracted from `nonRebasing` is to be fetched before the change to user balance accounting. We might quickly forget it